src/common/options.cc

```diff
@@ -6462,6 +6462,22 @@ std::vector<Option> get_rgw_options() {
     .set_description("Max number of keys to remove from garbage collector log in a single operation")
     .add_see_also({"rgw_gc_max_objs", "rgw_gc_obj_min_wait", "rgw_gc_processor_max_time", "rgw_gc_max_concurrent_io"}),
 
+    Option("rgw_gc_max_urgent_data_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(3072)
+    .set_description("maximum allowed size of urgent data in queue head for gc"),
+
+    Option("rgw_gc_max_queue_size", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
+    .set_default(128_M)
+    .set_description("Maximum allowed queue size for gc")
+    .set_long_description(
+        "The maximum allowed size of each gc queue, and its value should not "
+        "be greater than osd_max_object_size.")
+    .add_see_also({"osd_max_object_size"}),
+
+    Option("rgw_gc_num_urgent_data_entries", Option::TYPE_UINT, Option::LEVEL_ADVANCED)
```

Comment:
the term 'urgent data' is part of the cls_queue abstraction. i think a name like `rgw_gc_max_deferred` would be easier for users to understand

Reply:
yes, ok

Reply:
Should this - 'rgw_gc_max_urgent_data_size' also be changed to point to reflect that is related to deferred entries?

Reply:
yeah, the two variables should be consistent with each other